x-pack/plugins/file_upload/public/components/geojson_upload_form/index_name_form.tsx

```diff
@@ -0,0 +1,164 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { ChangeEvent, Component } from 'react';
+import { EuiFormRow, EuiFieldText, EuiCallOut, EuiSpacer } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import {
+  getExistingIndexNames,
+  getExistingIndexPatternNames,
+  checkIndexPatternValid,
+  // @ts-expect-error
+} from '../../util/indexing_service';
+
+export interface Props {
+  indexName: string; // Allow setting index pattern from external source
+  onIndexNameChange?: (name: string, error?: string) => void;
+}
+
+interface State {
+  existingIndexNames: string[];
+  indexName: string;
+  indexNameError: string | undefined;
+}
+
+export class IndexNameForm extends Component<Props, State> {
+  private _isMounted = false;
+
+  state: State = {
+    existingIndexNames: [],
+    indexName: '',
+    indexNameError: '',
+  };
+
+  async componentDidMount() {
+    this._isMounted = true;
+    this._loadExistingIndexNames();
+  }
+
+  componentWillUnmount() {
+    this._isMounted = false;
+  }
+
+  componentDidUpdate(prevProps: Props) {
```

Comment:
This can be removed once all state is only stored in a single place and always pushed down.

Reply:
We still want this even with the updated external state handling. Name updates get pushed down to this component from `GeoJsonUploadForm` and then pass through error handling in `IndexNameForm` which is called from here